examples/dart/test/serialization_test.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:test/test.dart';
+import 'package:realm_dart/realm.dart';
+import 'utils.dart';
+
+import './pet.dart';
+
+main() {
+  test('serialize', () {
+
+    final config = Configuration.local([Pet.schema]);
+    final realm = Realm(config);
+
+    // create pet spider object
+    final spider = Pet('Jumping Spider', 8, DateTime.utc(2024, 4, 10));
+
+    realm.write(() {
+      realm.add(spider);
+    });
+
+    // :snippet-start: serialize
+    // Pass the object as a parameter to the method
+    EJsonValue serializeByParam = toEJson(spider);
+
+    // Call the method directly on the object
+    EJsonValue serializeWithCall = spider.toEJson();
+    // :snippet-end:
+
+    print(serializeByParam);
+
+    final birthDate = DateTime.utc(2024, 4, 10);
+
+    // make sure serialized value matches expected
+    expect(serializeByParam, {
+      'type': 'Jumping Spider',
+      'numberOfLegs': {'\$numberInt': '8'},
+      'birthDate': {
+        '\$date': {'\$numberLong': birthDate.millisecondsSinceEpoch.toString()}
+      },
+      'price': null,
+    });
+
+    // make sure two methods of serialization match
+    expect(serializeByParam, serializeWithCall);
+
+    cleanUpRealm(realm);
+  });
+
+  test('deserialize', () {
+
+    final config = Configuration.local([Pet.schema]);
+    final realm = Realm(config);
+
+    // create pet spider object
+    final spider = Pet('Jumping Spider', 8, DateTime.utc(2024, 4, 10));
+
+    realm.write(() {
+      realm.add(spider);
+    });
+
+    // Pass the object as a parameter to the method
+    EJsonValue serializeByParam = toEJson(spider);
+
+    // :snippet-start: deserialize
+    // Pass the serialized object to the method
+    Pet deserializeFromEjson = fromEJson(serializeByParam);
```

Comment:
```suggestion
    final deserializeFromEjsonWithExplicitType = fromEJson<Pet>(serializeByParam);
    // or the shorter from where type is inferred ..
    Pet deserializeFromEjson = fromEJson(serializeByParam);
```
I know I suggested and prefer the latter form, but perhaps it is good to draw attention to the fact that `fromEjson<T>` is a generic function 🤔